### PR TITLE
Task/proof read

### PR DESCRIPTION
**Pull request by kzangeli**
Proof-read and more of the implementation documentation

### CODE REVIEW

File: doc/manuals/devel/README.md [hunk 1]
```diff
@@ -109,35 +107,36 @@ The log message library takes care of writing log lines to the log file, whose d
 `/tmp/contextBroker.log` which can be modified using the [CLI parameter](../admin/cli.md) `-logDir`.   The most important task, apart from helping in debugging the broker, are the transaction tags.
 
 See `lmTransactionStart()` et al in `logMsg.cpp/h`.
-About debugging the broker, see especially the trace levels (that are set using the `-t` CLI param),
-in `traceLevels.h`.
+About debugging the broker, see especially the trace levels (that are set using the `-t` CLI param), in `traceLevels.h`.
+And. to turn on traces, remember that it is not enough to set the trace levels. The option `-logLevel DEBUG` must be used as well.
```
Comment (fgalan) [inline doc/manuals/devel/README.md]: Typo

File: doc/manuals/devel/README.md [hunk 2]
```diff
@@ -187,58 +186,58 @@ upon reception of connections. This function is called `connectionTreat()` and i
...
 
 Internally, the broker keeps a vector inside a struct of the type `ConnectionInfo` for the HTTP headers
 and another vector for the URI parameters.
 
 For detailed information about the microhttpd library, see [its dedicated page in gnu.org](https://www.gnu.org/software/libmicrohttpd/).
 
-### Request Flow
+### Request flow
 
 <a name='flow-rq-01'></a>
```
Comment (fgalan) [inline doc/manuals/devel/README.md]: Overal  `name='...'` to `name="..."` change missing.

File: doc/manuals/devel/cprs.md [hunk 3]
```diff
@@ -34,14 +35,16 @@ Note that there are a number of service routines that end up calling `postUpdate
 
 * All attributes in the incoming payload are marked as **Not Found** (step 1).
 * [**mongoBackend** library](README.md#srclibmongobackend) processes the request (see diagrams [MB-01](mongoBackend.md#flow-mb-01) or [MB-02](mongoBackend.md#flow-mb-02)) and marks all attributes in the requests in one of three possible ways (step 2):
-       * Not Found
-       * Found in Local Broker
-       * Found in Remote Context Provider
-* The attributes that are found in a remote context provider need to be forwarded. The local attributes are simply updates while those not found are marked as such in the response.
-* A new vector of `ContextElementResponse` is created and filled with all those attributes that are to be forwarded (step 3). These responses are then added to the response vector that was output from **mongoBackend**. If no attribute is "found", then the `ContextElementResponse` is prepared with a 404 Not Found.
+
+    * Not Found
```
Comment (fgalan) [inline doc/manuals/devel/cprs.md]: Have you tested this render correctly in github web? List within list are tricky in markdown...

File: doc/manuals/devel/jsonParse.md [hunk 4]
```diff
@@ -12,11 +12,13 @@ Orion Context Broker contains not one but **two** libraries for JSON parsing. Th
 
 This document describes NGSIv1 parsing details. NGSIv2 parsing details are described in a [separate document](jsonParseV2.md).
 
-In general, the NGSIv1 parsing logic is more complex than NGSIv2 logic. Good news are that you will probably don't need to change anything in NGSIv1 parsing as this is the old version of the Orion API and the work should now concentrate in evolving NGSIv2, not NGSIv1.
+In general, the NGSIv1 parsing logic is more complex than NGSIv2 logic. The good news is that you will probably not need to change anything in NGSIv1 parsing as this is the old version of the Orion API and the work should now concentrate in evolving NGSIv2, not NGSIv1.
 
-The purpose of the parse step is to transform a text buffer, JSON is this case, to an instance of a class/struct in C++. A very simple but illustrative example:
+The purpose of the parse step is to transform a text buffer, JSON in this case, to an instance of a class/struct in C++. The external library takes care of parsing the JSON string while the Orion librfary extracts the information and populates the approipriate instance of the class/struct in C++.
```
Comment (fgalan) [inline doc/manuals/devel/jsonParse.md]: Typo (librfary)

File: doc/manuals/devel/mongoBackend.md [hunk 5]
```diff
@@ -168,117 +169,117 @@ Case 6: action type is "DELETE" to remove an entity
...
+    * If subscription cache is being used, then the subscription is retrieved from the subscription cache calling `subCacheItemLookup()` (step 7). Next, last notification time and count are modified in the subscription cache (they will be consolidated in the database in the next subscription cache refresh, see details in [this document](subscriptionCache.md#subscription-cache-refresh)). The access to the subscription cache is protected by the subscription cache semaphore (see [this document for details](semaphores.md#subscription-cache-semaphore)), which is taken and released in steps 6 and 8 respectively.
+
+Finally, in the case of action type "UPDATE/REPLACE", the context update logic is able to "fill the gaps" for missing entities/attributes in the local database with Context Provider information. This is done in `searchContextProviders()`. The detail is shown in the sequence diagram below.
 
 <a name='flow-md-02'></a>
 ![`searchContextProviders()` function detail](images/Flow-MD-02.png)
 
-_MD-02: `searchContextProviders()` function detail_
+_MD-02: `searchContextProviders()` function detail_-
```
Comment (fgalan) [inline doc/manuals/devel/mongoBackend.md]: I think this is a typo. The final `-` should be removed.

Are the any other similar case? A grep for `_-` could detect them...

File: doc/manuals/devel/mongoBackend.md [hunk 6]
```diff
@@ -287,13 +288,13 @@ The detail for `mongoEntityTypes()` is as shown in the following diagram.
...
-* Depending on `-reqMutexPolicy`, request semaphore could be taken (read mode) (step 2). See [this document for details](semaphores.md#mongo-request-semaphore). 
-* A list of entity types and attributes belonging each type is retrieved from DB, using the `runCollectionCommand()` function in the `connectionOperations` module to run an aggregation command (steps 3 and 4).
+* `mongoEntityTypes()` is invoked from a service routine (step 1). This can be from either `getEntityTypes()` (which resides in `lib/serviceRoutines/getEntityTypes.cpp`) or `getEntityAllTypes()` (which resides in `lib/serviceRoutinesV2/getEntityAllTypes.cpp`).
+* Depending on `-reqMutexPolicy`, the request semaphore may be taken (read mode) (step 2). See [this document for details](semaphores.md#mongo-request-semaphore). 
+* A list of entity types and attributes ?belonging to? each type is retrieved from the database, using `runCollectionCommand()` in the `connectionOperations` module, to run an aggregation command (steps 3 and 4).
```
Comment (fgalan) [inline doc/manuals/devel/mongoBackend.md]: What ?...? means here?

File: doc/manuals/devel/mongoBackend.md [hunk 7]
```diff
@@ -168,117 +169,117 @@ Case 6: action type is "DELETE" to remove an entity
...
+* Then, `fillContextProviders()` (in the `MongoGlobal` module) is called to attempt to fill the not found attributes with the matching registrations (step 5).
+* If at least one attribute still has the `found` flag set to `false`, a new lookup round is done. This time, searching for whole entities (i.e. in the "E-&lt;null&gt;" form). Again, `registrationsQuery()` is used (step 6). This function searches the database using `collectionRangedQuery()` in the `connectionOperations` module (steps 7 and 8).
+* Then, `fillContextProviders()` (in the `MongoGlobal` module) is called again to attempt to fill the not found attributes with the new matched registrations (step 9).
 
 [Top](#top)
 
 #### `mongoQueryContext` (SR)
 
-Encapsulates the logic for the query context operation.
+`mongoQueryContext()` encapsulates the logic for the query context operation.
```
Comment (fgalan) [inline doc/manuals/devel/mongoBackend.md]: I think that here it refers to the module name, not the function. Thus, `()` shouldn't be used.

Maybe are other case of this "anti-style"? Plase, review it globaly in mongoBackend.md

File: doc/manuals/devel/mongoBackend.md [hunk 8]
```diff
@@ -587,94 +593,94 @@ typedef struct MongoConnection
...
+* `processConditionVector()` (step 1) is invoked by mongoBackend functions. See diagrams [MB-11](#flow-mb-11) and [MB-12](#flow-mb-12).
+* A loop iterates over each individual condition in the `NotifyConditionVector` vector (although most of the times this vector has only one item):
+   * `processOnChangeConditionForSubscription()` is called to process the individual condition (step 2).
+   * `entitiesQuery()` is called to get the entities to be included in the notification (step 3), which in sequence relies on `collectionRangedQuery()` in the `connectionOperations` module in order to get the entities from the database (steps 4 and 5).
+   * `pruneContextElements()` is called in order to remove not found elements, as it makes no sense including them in the notification (step 6).
+   * If, after pruning, there is any entity to send, steps 7 to 11 are executed.
```
Comment (fgalan) [inline doc/manuals/devel/mongoBackend.md]: Again, check that list-within-list is correctly rendered at github web.

File: doc/manuals/devel/cprs.md (hunk 3 above)
Comment (kzangeli) [inline doc/manuals/devel/cprs.md]: True, haven't done that yet

File: doc/manuals/devel/mongoBackend.md (hunk 7 above)
Comment (kzangeli) [inline doc/manuals/devel/mongoBackend.md]: True

File: doc/manuals/devel/mongoBackend.md (hunk 6 above)
Comment (kzangeli) [inline doc/manuals/devel/mongoBackend.md]: Yeah ... look at the original:

A list of entity types and attributes belonging each type ...

The phrase is not complete. I don't know how to modify it, that's the reason for the ??

File: doc/manuals/devel/mongoBackend.md [hunk 9]
```diff
@@ -28,556 +28,562 @@
...
+Note the usage of the `noAttrDetails` parameter in `mongoEntityTypes()` and `mongoAttributesForEntityType()` in order to avoid a (potentially costly) process to get types of the attributes associated to an entity type (implemented by `getAttributeTypes()`).
 
-All the above functions heavily relies in the MongoDB aggregation framework. You should be familiar with this framework (and with the `entities` collection structure, [described as part of the database model in the administration documentation](../admin/database_model.md#entities-collection)) in order to understand how they work.
+All the above functions heavily rely on the MongoDB aggregation framework. You should be familiar with this framework (and with the `entities` collection structure, [described as part of the database model in the administration documentation](../admin/database_model.md#entities-collection)) in order to understand how the functions work.
 
 [Top](#top)
 
 #### `mongoCreateSubscription` (SR2)
 
-Encapsulates the contex
```
Comment (kzangeli) [inline doc/manuals/devel/mongoBackend.md]: Here is another unfinished phrase.

Original: Among them (details in source code) we highly `setNotification()` (step 5), due to ...


We highly ... what?

Comment (kzangeli) [inline doc/manuals/devel/mongoBackend.md]: One more "highly what?"
A verb is missing ...

File: doc/manuals/devel/cprs.md (hunk 3 above)
Comment (kzangeli) [inline doc/manuals/devel/cprs.md]: Ir looks good in https://github.com/telefonicaid/fiware-orion/blob/task/proof_read/doc/manuals/devel/cprs.md

File: doc/manuals/devel/mongoBackend.md (hunk 8 above)
Comment (kzangeli) [inline doc/manuals/devel/mongoBackend.md]: Looks good in https://github.com/telefonicaid/fiware-orion/blob/task/proof_read/doc/manuals/devel/mongoBackend.md

File: doc/manuals/devel/README.md (hunk 1 above)
Comment (kzangeli) [inline doc/manuals/devel/README.md]: Fixed in 48f051122

File: doc/manuals/devel/jsonParse.md (hunk 4 above)
Comment (kzangeli) [inline doc/manuals/devel/jsonParse.md]: Fixed in 48f0511

File: doc/manuals/devel/mongoBackend.md (hunk 5 above)
Comment (kzangeli) [inline doc/manuals/devel/mongoBackend.md]: Fixed in 48f0511.
`grep` done, no more _-

File: doc/manuals/devel/mongoBackend.md (hunk 7 above)
Comment (kzangeli) [inline doc/manuals/devel/mongoBackend.md]: Fixed in 48f0511

**Pull request merged**